src/containers/Dashboard/DashboardContainer.tsx

```diff
@@ -13,28 +13,40 @@ import {useSelector, useDispatch} from "react-redux";
 import {RootState} from "../../app/rootReducer";
 import {useHistory} from "react-router-dom";
 import {Routes} from "../../constants/routes";
-import {getNodeList} from "./NodeSlice";
+import {getNodeList, getDiskInfo, getGeneralInfo} from "./NodeSlice";
 
 export const DashboardContainer = (): ReactElement => {
-    const [areElementsHidden, setElementsHidden] = useState(true);
+    const [areElementsHidden, setElementsHidden] = useState<boolean>(true);
+    const [selectedNodeIndex, setSelectedNodeIndex] = useState<number>(0);
     const history = useHistory();
-    const state = useSelector((state: RootState) => state);
     const dispatch = useDispatch();
+    const state = useSelector((state: RootState) => state);
+    const token = state.user.token;
+    const nodeList = state.node.nodeList;
```

Comment:
Why you have `nodeList` and `nodeListComplete` ? When can a list be incomplete?

Reply:
"nodeListComplete" is boolean type, and reducer sets it to true when fetch is complete
its only use is to re-run useEffect to show statistic of 1. node by default, 
if there are no nodes page shows no nodes and add button

"nodeList" can't be condition, as it changes and re-runs useEffect it creates infinite loop

I'm not sure, is there better way of handling this case?

Reply:
Didn't catch it - means it needs to be better :)

Yes, you need the flag for fetching but let it have a better name. You can use `isLoading` for this and immediately have loading indicator info displayed to user (nothing fancy for the moment).